src/components/signin/components/Signin.js

```diff
@@ -35,16 +42,34 @@ class Signin extends React.Component {
                 password,
             };
             signinRequest(payload);
-            this.setState({ visible: false });
         });
-
     }
 
     saveFormRef = formRef => {
         this.formRef = formRef;
     }
 
+    componentDidUpdate(){
+        const { 
+            isSuccessful, 
+            actions: { resetSignState }, 
+            signinfailMessage, 
+            signinError,
+        } = this.props;
+        const { visible } = this.state;
+    
+        if (visible && isSuccessful) {
+            this.setState({ visible: false });
+            resetSignState();
+        }
+        if (signinError) {
+            message.error(signinfailMessage,3);
+            resetSignState();
+        }
```

Comment:
there ideally should never be a situation where you sign-in successfully also have a sign-in failure. So the the check of `signinError` should be an else if statement. This is a slight optimization of your code
```
   ...
   if (visible && isSuccessful {
       ...
   } else if (signinError) {
       ...
   }
```